src/lib/sap_ha/semantic_checks.rb

```diff
@@ -235,8 +239,7 @@ def integer_in_range(value, low, high, message = '', field_name = '')
     def sap_sid(value, message = '', field_name = '')
       message = "A valid SAP System ID consists of three characters, starts with a letter, and "\
       " must not collide with one of the reserved IDs" if message.nil? || message.empty?
-      flag = !SAP_SID_REGEXP.match(value).nil?
-      flag &= !RESERVED_SAP_SIDS.include?(value)
+      flag = !SAP_SID_REGEXP.match?(value).nil? && !RESERVED_SAP_SIDS.include?(value)
```

Comment:
If you decide to use `match?`, you can drop the `!` and the `nil?`. See https://github.com/yast/yast-sap-ha/pull/9#discussion_r942158836.